File: fsharp/bob/Bob.fs
module Bob
open System

let isUpper str = 
    match str |> Seq.filter Char.IsLetter |> Seq.toList with
    | [] ->  false
    | x  -> x |> Seq.forall Char.IsUpper

let (|IsYelling|WithoutSayingAnything|AnythingElse|) (str:string) = 
    match str with
    | _ when String.IsNullOrWhiteSpace(str) -> WithoutSayingAnything
    | _ when isUpper str                    -> IsYelling
    | _                                     -> AnythingElse

let (|IsQuestion|_|) (str:string) = 
    match str.Trim().EndsWith "?" with
    | true  -> Some IsQuestion
    | false -> None

let response (input: string): string = 
    match input with
    | WithoutSayingAnything  -> "Fine. Be that way!"
    | IsYelling & IsQuestion -> "Calm down, I know what I'm doing!"
    | IsQuestion             -> "Sure." 
    | IsYelling              -> "Whoa, chill out!"
    | AnythingElse           -> "Whatever."

File: csharp/bob/Bob.cs
using System;

public static class Bob
{
    public static string Response(string statement)
    {
        if(statement.Trim().Length == 0 || statement.Trim().Equals(""))
        {
            return "Fine. Be that way!";
        }

        int countIsLetter = 0;
        int countIsUpper = 0;

        for(int i = 0; i < statement.Length; i++ )
        { 
            if(char.IsUpper(statement[i]))
            {
                countIsUpper++;

            }
            if(char.IsLetter(statement[i]))
            {
                countIsLetter++;
            }
        }
        if (countIsUpper > 0 && countIsLetter > 0 && countIsUpper == countIsLetter)
        {
            return "Whoa, chill out!";
        }
        
        if(statement.Trim().EndsWith("?"))
        {
            return "Sure.";
        }
        return "Whatever.";
    }
}